### PR TITLE
Flashlight spawner

### PR DESCRIPTION
**Pull request by RebeccaStankus**
TEST PLAN
- Put this server script on any collisionless, non-grabbable item (spawner item).
- A flashlight should appear that can be grabbed
- When the flashlight is moved more than 0.1m from the position of the spawner item, a new one appears within 5 seconds.

### CODE REVIEW

File: DomainContent/Halloween/flashlightSpawner.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            _this.checkingStatus();
+        },
+
+        checkingStatus: function() {
+            checking = Script.setInterval(function() {
+                var needClone = true;
+                Entities.findEntities(_this.position, SEARCH_RADIUS).forEach(function(element) {
+                    var name = Entities.getEntityProperties(element, 'name').name;
+                    if (name === "Flashlight") {
+                        print("found another flashlight ", element, " at ", JSON.stringify(
```
Comment (misslivirose) [inline DomainContent/Halloween/flashlightSpawner.js]: Debug print

File: DomainContent/Halloween/flashlightSpawner.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                lifetime: 1800,
+                dimensions: {
+                    x: 0.374295175075531,
+                    y: 0.2721865177154541,
+                    z: 0.24757488071918488
+                },
+                visible: true,
+                modelURL: "https://hifi-content.s3.amazonaws.com/jimi/environment/201804_tomb/BuildersTorch.fbx?1?1",
+                name: "Flashlight",
+                serverScripts: Script.resolvePath("emptyServerScript.js"),
```
Comment (misslivirose) [inline DomainContent/Halloween/flashlightSpawner.js]: Should this be in the PR next to this script?

Comment (RebeccaStankus) [reply DomainContent/Halloween/flashlightSpawner.js]: oh, yes

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete.

Put this server script on any collisionless, non-grabbable item (spawner item). **Pass**
A flashlight should appear that can be grabbed **Pass**
When the flashlight is moved more than 0.1m from the position of the spawner item, a new one appears within 5 seconds. **Pass**

All **Pass** :)

**Comment by AndySeattle** [comment]
@misslivirose @RebeccaStankus A few questions with this PR. While the flashlights are re-spawning / you can grab them, i am not seeing them with a "put down decay" timer, so could we end up with a domain full of flash lights? also they do not attach to the player light an avatar item (see the Material gun or Rainbow Cannon in the spot.) like i would expect? So people on desktop can NOT use them and you have to hold on to them the entire time you use them. 

Is this intentional? or should i expect them to function like the material gun in the spot?

**Comment by AndySeattle** [comment]
@RebeccaStankus did we get the scripts adjusted much like you did with the rainbow gun in the spot for the flashlights in haunted? Think im going to send this Salinas way if you have?

**Comment by misslivirose** [comment]
@AndySeattle - The forest domain isn't intended to be a desktop domain given the darkness anyway - I think these are a pass given that they're deployed in there already and have lifetimes on them. Given that we're more than halfway through October I'd like to merge this.

**Pull request merged**